.circleci/config.yml

```diff
@@ -84,20 +84,34 @@ workflows:
   version: 2
   build:
     jobs:
-      - checkout_code
+      - checkout_code:
+          filters:
+            tags:
+              only: /release.*/
       - install-dependencies:
           requires:
             - checkout_code
+          filters:
+            tags:
+              only: /release.*/
       - test:
           requires:
             - install-dependencies
+          filters:
+            tags:
+              only: /release.*/
       - build:
           requires:
             - install-dependencies
+          filters:
+            tags:
+              only: /release.*/
       - publish:
           requires:
             - build
           filters:
             branches:
               only:
                 - master
+            tags:
+              only: /release.*/
```

Comment:
unfortunately CircleCI treats `tags` and `branches` separately. I'm afraid we cannot say IF branch=master AND tag=release.* then act. Which means that if we have a match on a tag then we will fire off `publish` job.

an example:
https://circleci.com/workflow-run/a3bd9483-36f8-42a2-8fd8-c97a51b92cf7

Reply:
Which means that `publish` can be triggered on any branch not just master.

Reply:
So this code means, we only publish when the tag begins with `release.` ?

Reply:
yes, but also on branch=master push (as before)

Reply:
The prefix should only be `v` as we'll only tag with the version.
For example:
- `v0.2.1`
- `v0.3.0`
etc.

Reply:
we could however read a branch name directly in `publish` job and fail if there is no match.